File: app/src/main/java/nz/co/warehouseandroidtest/WarehouseService.java
package nz.co.warehouseandroidtest;

import java.util.Map;

import nz.co.warehouseandroidtest.data.ProductDetail;
import nz.co.warehouseandroidtest.data.SearchResult;
import nz.co.warehouseandroidtest.data.User;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface WarehouseService {

    @GET("bolt/newuser.json")
    Call<User> getNewUserId();

    @GET("bolt/price.json")
    Call<ProductDetail> getProductDetail(@QueryMap Map<String, String> paramMap);

    @GET("bolt/search.json")
    Call<SearchResult> getSearchResult(@QueryMap Map<String, String> paramMap);
}

File: app/src/main/java/nz/co/warehouseandroidtest/WarehouseService.kt
package nz.co.warehouseandroidtest

import nz.co.warehouseandroidtest.data.ProductDetail
import nz.co.warehouseandroidtest.data.SearchResult
import nz.co.warehouseandroidtest.data.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.QueryMap
import java.util.HashMap

interface WarehouseService {
    @GET("bolt/newuser.json")
    fun getNewUserId(): Call<User?>?

    @GET("bolt/price.json")
    fun getProductDetail(@QueryMap paramMap: HashMap<String, String>): Call<ProductDetail>

    @GET("bolt/search.json")
    fun getSearchResult(@QueryMap paramMap: Map<String, String>): Call<SearchResult>
}